docs/specs/user-import.md

```diff
@@ -0,0 +1,233 @@
+- `email_verified`, `phone_number_verified`: The update behavior is **UPDATED_IF_PRESENT**. For example, in the first import, if `email_verified` is absent, then email is marked as unverified.
+- The update behavior of other standard attributes is **UPDATED_IF_PRESENT_AND_REMOVED_IF_NULL**. In particular, `address` IS NOT merged with the existing value, but REPLACES the existing `address` value.
+- `custom_attributes.*`: For each attribute in `custom_attributes`, the update behavior is **UPDATED_IF_PRESENT_AND_REMOVED_IF_NULL**.
+- `roles` and `groups`: The update behavior is **UPDATED_IF_PRESENT`.
+- `disabled`: The update behavior is **UPDATED_IF_PRESENT**. So re-importing a user without specifying `disabled` WILL NOT accidentally alter the disabled state previously set by other means.
+- `password`: The update behavior is **IGNORED**. If `password` was not provided when the user was first imported, subsequent import CANNOT add password.
+- `mfa.email`: The update behavior is **UPDATED_IF_PRESENT_AND_REMOVED_IF_NULL**. The user can perform 2FA with email OTP.
+- `mfa.phone_number`: The update behavior is **UPDATED_IF_PRESENT_AND_REMOVED_IF_NULL**. The user can perform 1FA with phone OTP.
+- `mfa.password`: The update behavior is **IGNORED**. If `mfa.password` was not provided when the user was first imported, subsequent import CANNOT add additional password.
+- `mfa.totp`: The update behavior is **IGNORED**. If `mfa.totp` was not provided when the user was first imported, subsequent import CANNOT add TOTP.
+
+## Update behavior
+
+- **UPDATED_IF_PRESENT_AND_REMOVED_IF_NULL**: If the field is present and the value is not null, then it is updated. If the field is present and the value is null, then it is removed. If the field is absent, no operation is done.
+- **UPDATED_IF_PRESENT**: If the field is present, then it is updated. If the field is absent, no operation is done.
+- **IGNORED**: If the user exists already, then the field is ignored. If the field is absent, no operation is done.
+
+## Supported password format
+
+### Plain password
+
+```
+{
+  "type": "plain",
+  "plain_password": "secret"
+}
+```
+
+### Bcrypt password
+
+```
+{
+  "type": "bcrypt",
+  "password_hash": "$2a$10$N9qo8uLOickgx2ZMRZoMyeIjZAgcfl7p92ldGxad68LJZdL17lhWy"
+}
+```
+
+## The response
+
+You will receive a response similar to the following when you just initiated an import.
+
+```
+{
+  "id": "some_opaque_string",
+  "created_at": "2024-01-01T00:00:00.000Z"
+  "status": "pending",
+}
+```
+
+You will receive a response similar to the following when the import completed (with or without errors).
+Sensitive values are redacted in the response.
+
+```
+{
+  "id": "some_opaque_string",
+  "created_at": "2024-01-01T00:00:00.000Z"
+  "status": "completed",
+  "summary": {
+    "total": 100,
+    "inserted": 50,
+    "updated": 49,
+    "failed": 1
+  },
+  "errors": [
```

Comment:
Can we include the item index here, so user can easily find the item?

Reply:
JSON path seems overkill. I added `index` instead.